src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoConditionalExpressionsTest.java

```diff
@@ -0,0 +1,206 @@
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+@Category(PythonTest.class)
+public class DjangoConditionalExpressionsTest extends DjangoTestSetup {
+
+  @Parameter public String host;
+
+  @Parameters(name = "host = {0}")
+  public static List<Object[]> data() {
+    return ImmutableList.of(new Object[] {"localhost"}, new Object[] {"/tmp"});
+  }
+
+  private ResultSet createResultSet(List<String> rows, int numCols) {
+    ResultSet.Builder resultSetBuilder = ResultSet.newBuilder();
+
+    StructType.Builder builder = StructType.newBuilder();
+
+    for (int i = 0; i < 2 * numCols; i += 2) {
+      if (rows.get(i).equals("int")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(Type.newBuilder().setCode(TypeCode.INT64).build())
+                .setName(rows.get(i + 1))
+                .build());
+
+      } else if (rows.get(i).equals("string")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(Type.newBuilder().setCode(TypeCode.STRING).build())
+                .setName(rows.get(i + 1))
+                .build());
+      }
+    }
+    resultSetBuilder.setMetadata(
+        ResultSetMetadata.newBuilder().setRowType(builder.build()).build());
+    for (int i = 2 * numCols; i < rows.size(); i += numCols) {
+      ListValue.Builder rowBuilder = ListValue.newBuilder();
+      for (int j = i; j < i + numCols; ++j)
+        rowBuilder.addValues(Value.newBuilder().setStringValue(rows.get(j)).build());
+
+      resultSetBuilder.addRows(rowBuilder.build());
+    }
+    return resultSetBuilder.build();
+  }
+
+  @Test
+  public void testConditionalUpdate() throws IOException, InterruptedException {
+    String sqlUpdate =
+        "UPDATE \"singers\" SET \"firstname\" = CASE WHEN (\"singers\".\"firstname\" = 'hello') THEN 'h' WHEN (\"singers\".\"firstname\" = 'world') THEN 'w' ELSE 'n' END";
+
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(sqlUpdate), 17));
+
+    String expectedOutput = "17\n";
+    List<String> options = new ArrayList<>();
+    options.add("update");
+    String actualOutput = executeConditionalTests(pgServer.getLocalPort(), host, options);
+    assertEquals(expectedOutput, actualOutput);
+
+    assertEquals(1, mockSpanner.countRequestsOfType(ExecuteSqlRequest.class));
+    assertEquals(sqlUpdate, mockSpanner.getRequestsOfType(ExecuteSqlRequest.class).get(0).getSql());
+  }
+
+  @Test
+  public void testConditionalAggregation() throws IOException, InterruptedException {
+    String sqlSelect =
+        "SELECT COUNT(\"singers\".\"singerid\") FILTER (WHERE \"singers\".\"firstname\" = 'hello') AS \"hello_count\", COUNT(\"singers\".\"singerid\") FILTER (WHERE \"singers\".\"firstname\" = 'world') AS \"world_count\" FROM \"singers\"";
+
+    List<String> result = new ArrayList<>();
+
+    result.add("int");
```

Comment:
Here and elsewhere: Could we use a more appropriate data structure for this instead of a list with specific keywords? I understand the need to use something like a simple list when sending the data to Python, but as far as I can tell, this list is only used in the Java test code.

Reply:
Very Good Point. Since, these are just unit tests, I didn't think much about creating some proper class for this case. I am merging this PR for now. But in future, I'll definitely refactor this part to look better